# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/mediacontrol/MediaControlUtils.h
# Commit: 7333d4f4c7d1
# Full Hash: 7333d4f4c7d1b5ad585d222157067aa4276f9945
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-09-10 03:29:56
# Description:
#   Bug 1579437 - Check for discarded BrowsingContext in more places, r=farre
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45088
# ==============================================================================

diff -r 941fe2efca5d -r 7333d4f4c7d1 dom/media/mediacontrol/MediaControlUtils.h
--- a/dom/media/mediacontrol/MediaControlUtils.h	Mon Sep 09 17:30:08 2019 +0000
+++ b/dom/media/mediacontrol/MediaControlUtils.h	Mon Sep 09 12:04:13 2019 +0000
@@ -63,7 +63,7 @@
 
 void NotifyMediaStarted(uint64_t aWindowID) {
   RefPtr<BrowsingContext> bc = GetBrowingContextByWindowID(aWindowID);
-  if (!bc) {
+  if (!bc || bc->IsDiscarded()) {
     return;
   }
   LOG("Notify media started in BC %" PRId64, bc->Id());
@@ -73,7 +73,7 @@
 
 void NotifyMediaStopped(uint64_t aWindowID) {
   RefPtr<BrowsingContext> bc = GetBrowingContextByWindowID(aWindowID);
-  if (!bc) {
+  if (!bc || bc->IsDiscarded()) {
     return;
   }
   LOG("Notify media stopped in BC %" PRId64, bc->Id());
@@ -83,7 +83,7 @@
 
 void NotifyMediaAudibleChanged(uint64_t aWindowID, bool aAudible) {
   RefPtr<BrowsingContext> bc = GetBrowingContextByWindowID(aWindowID);
-  if (!bc) {
+  if (!bc || bc->IsDiscarded()) {
     return;
   }
   LOG("Notify media became %s in BC %" PRId64,
